File: 53-maximum-subarray/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        subArr = nums[0]
        maxSub = nums[0]

        for i in range(1,len(nums)):
            subArr = max(nums[i], nums[i]+subArr)
            maxSub = max(maxSub, subArr)

        return maxSub

File: 53-maximum-subarray/maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int sub = nums[0];
        int maxSub = nums[0];

        for(int i = 1; i < nums.size(); i++){
            sub = max(sub + nums[i], nums[i]);
            maxSub = max(maxSub, sub);
        }

        return maxSub;
    }
};